extension/webviews/components/SearchViewer.svelte

```diff
@@ -0,0 +1,65 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import type { Thread as ThreadType } from '../types';
+  import { Page } from '../enums';
+  import { currentProject, page } from '../state/store';
+  import Thread from './thread/Thread.svelte';
+  import Button from './Button.svelte';
+
+  let searchQuery: string = '';
+  let threads: Array<ThreadType> = [];
+
+  const handleSearchQuery = (event: any) => {
+    if (event.key === 'Enter') {
+      searchQuery = event.target.value;
+      if ($currentProject) {
+        tsvscode.postMessage({
+          type: 'searchThreads',
+          value: { searchText: searchQuery, projectId: $currentProject.id },
+        });
+      }
+    }
+  };
+
+  const handleCancelButtonClick = () => {
+    searchQuery = '';
+    $page = Page.ThreadsViewer;
+  };
+
+  const messageEventListener = async (event: any) => {
```

Comment:
I updated global.d.ts on https://github.com/doclin-dev/doclin/pull/136

We should be able to use Event type once the PR is merged.

Reply:
cool will update it once that PR is merged. because now, if I use `Event` then `message.data` would throw an error.